userspace/libsinsp/sinsp_events.h

```diff
@@ -0,0 +1,430 @@
+ */
+bool is_unused_event(ppm_event_code event_type);
+
+/**
+	 * @brief If the event type has one of the following flags return true:
+	 * - `EF_SKIPPARSERESET`
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has at least one of these flags.
+ */
+bool is_skip_parse_reset_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event has the `EF_OLD_VERSION` flag
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EF_OLD_VERSION` flag.
+ */
+bool is_old_version_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_SYSCALL` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_SYSCALL` category.
+ */
+bool is_syscall_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_TRACEPOINT` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_TRACEPOINT` category.
+ */
+bool is_tracepoint_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_METAEVENT` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_METAEVENT` category.
+ */
+bool is_metaevent(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_UNKNOWN` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_UNKNOWN` category.
+ */
+bool is_unknown_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_PLUGIN` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_PLUGIN` category.
+ */
+bool is_plugin_event(ppm_event_code event_type);
+
+/*=============================== Events related ===============================*/
+
+/*=============================== PPM_SC set related (sinsp_events_ppm_sc.cpp) ===============================*/
+
+/*!
+	\brief Provide the minimum set of syscalls required by `libsinsp` state collection.
+
+	WARNING: without merging your ppm_sc set with the one provided by this method,
+ 	we cannot guarantee that `libsinsp` state will always be up to date, or even work at all.
+*/
+set<ppm_sc_code> sinsp_state_sc_set();
+
+/*!
+  \brief Enforce simple set of syscalls with all the security-valuable syscalls.
+  It has same effect of old `simple_consumer` mode.
+  Does enforce minimum sinsp state set.
+*/
+set<ppm_sc_code> enforce_simple_sc_set(set<ppm_sc_code> ppm_sc_set = {});
```

Comment:
Nope, this is the `enforce` one (i decided to leave a `enforce_simple_X_set` under each {tp,sc} categories; i think that a consumer should still be able to use `enforce_simple_sc_set(mySet)` and forget about anything else.

Reply:
right now I would keep them, let's say these are our suggested sets to use for libs consumers. Falco won't probably use these APIs anymore.

Reply:
Either way fine with me since we now generally adopt an approach of leveraging generic set operations. As we are now breaking everything anyways my preference would be to go all in and just return the simple_set etc in the sinsp API -> that way one consistent behavior and approach across the board.